server/src/tests/listing_apis.rs

```diff
@@ -0,0 +1,286 @@
+            )
+            .await
+            .unwrap();
+        assert_eq!(account_resources, vec![account_struct_tag()]);
+
+        let account_resources = ctx
+            .mv_list_resources(
+                address,
+                None,
+                None,
+                BlockNumberOrTag::Number(call_receipt.inner.block_number.unwrap()),
+            )
+            .await
+            .unwrap();
+        assert_eq!(
+            account_resources,
+            vec![
+                account_struct_tag(),
+                counter_struct_tag(address.to_move_address())
+            ]
+        );
+
+        ctx.shutdown().await;
+        Ok(())
+    })
+    .await
+}
+
+async fn deploy_counter_contract(
+    ctx: &mut TestContext<'static>,
+    signer: &PrivateKeySigner,
+) -> TransactionReceipt {
+    let address = signer.address();
+    let bytecode = crate::tests::integration::create_move_counter_contract_bytecode(address);
+    let mut tx = TxEip1559 {
+        chain_id: ctx.genesis_config.chain_id,
+        nonce: 0,
+        gas_limit: u64::MAX,
+        max_fee_per_gas: 0,
+        max_priority_fee_per_gas: 0,
+        to: TxKind::Create,
+        value: U256::ZERO,
+        access_list: Default::default(),
+        input: bytecode.into(),
+    };
+    let signature = signer.sign_transaction_sync(&mut tx).unwrap();
+    let signed_tx = TxEnvelope::Eip1559(tx.into_signed(signature));
+    ctx.execute_transaction(signed_tx).await.unwrap()
+}
+
+async fn call_counter_publish(
+    ctx: &mut TestContext<'static>,
+    signer: &PrivateKeySigner,
+) -> TransactionReceipt {
+    let address = signer.address();
+    let input = TransactionData::EntryFunction(EntryFunction::new(
+        ModuleId::new(address.to_move_address(), ident_str!("counter").into()),
+        ident_str!("publish").into(),
+        Vec::new(),
+        vec![
+            bcs::to_bytes(&MoveValue::Signer(address.to_move_address())).unwrap(),
+            bcs::to_bytes(&MoveValue::U64(7)).unwrap(),
+        ],
+    ));
+    let mut tx = TxEip1559 {
+        chain_id: ctx.genesis_config.chain_id,
+        nonce: 1,
+        gas_limit: u64::MAX,
+        max_fee_per_gas: 0,
+        max_priority_fee_per_gas: 0,
+        to: TxKind::Call(address),
+        value: U256::ZERO,
+        access_list: Default::default(),
+        input: input.to_bytes().unwrap().into(),
+    };
+    let signature = signer.sign_transaction_sync(&mut tx).unwrap();
+    let signed_tx = TxEnvelope::Eip1559(tx.into_signed(signature));
+    ctx.execute_transaction(signed_tx).await.unwrap()
+}
+
+async fn get_all_modules(ctx: &TestContext<'static>, address: AccountAddress) -> Vec<Identifier> {
+    let address = address.to_eth_address();
+    let mut all_modules = Vec::new();
+    loop {
+        let mut modules = ctx
+            .mv_list_modules(address, all_modules.last(), None, BlockNumberOrTag::Latest)
+            .await
+            .unwrap();
+        if modules.is_empty() {
```

Comment:
nit: Probably not needed optimization for test, less code is more readable I guess.